language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/builder/impl/CreateVariableCodeAction.java

```diff
@@ -0,0 +1,129 @@
+        boolean hasCustomInitFunction = false;
+        boolean isAsync = false;
+        if (refAtCursor.getbLangNode() instanceof BLangInvocation) {
+            hasDefaultInitFunction = symbolAtCursor instanceof BObjectTypeSymbol;
+            hasCustomInitFunction = symbolAtCursor instanceof BInvokableSymbol &&
+                    symbolAtCursor.name.value.endsWith("init");
+            isAsync = ((BLangInvocation) refAtCursor.getbLangNode()).isAsync();
+        }
+
+        // Find enclosing function node
+        BLangNode bLangNode = refAtCursor.getbLangNode();
+        BLangFunction enclosedFunc = null;
+        while (!(bLangNode instanceof BLangPackage)) {
+            if (bLangNode instanceof BLangFunction) {
+                enclosedFunc = (BLangFunction) bLangNode;
+                break;
+            }
+            bLangNode = bLangNode.parent;
+        }
+
+        return getCreateVariableCodeActions(context, uri, enclosedFunc, position, refAtCursor,
+                                            hasDefaultInitFunction, hasCustomInitFunction, isAsync);
+    }
+
+
+    private static List<CodeAction> getCreateVariableCodeActions(LSContext context, String uri,
+                                                                 BLangFunction enclosedFunc, Position position,
+                                                                 Reference referenceAtCursor,
+                                                                 boolean hasDefaultInitFunction,
+                                                                 boolean hasCustomInitFunction, boolean isAsync) {
+        List<CodeAction> actions = new ArrayList<>();
+
+
+        BLangNode bLangNode = referenceAtCursor.getbLangNode();
+        CompilerContext compilerContext = context.get(DocumentServiceKeys.COMPILER_CONTEXT_KEY);
+
+
+        List<TextEdit> importEdits = new ArrayList<>();
+        Pair<List<String>, List<String>> typesAndNames = getPossibleTypesAndNames(context, referenceAtCursor,
+                                                                                  hasDefaultInitFunction,
+                                                                                  hasCustomInitFunction, isAsync,
+                                                                                  bLangNode,
+                                                                                  importEdits, compilerContext);
+
+        List<String> types = typesAndNames.getLeft();
+        List<String> names = typesAndNames.getRight();
+
+        for (int i = 0; i < types.size(); i++) {
+            String type = types.get(i);
+            String name = names.get(i);
+            String commandTitle = CommandConstants.CREATE_VARIABLE_TITLE;
+            if (types.size() > 1) {
+                String typeLabel = (type.startsWith("[") && type.endsWith("]") && !type.endsWith("[]") &&
+                        type.length() > 10) ? "Tuple" : type;
```

Comment:
Shall we move this to a private util and is there a specific reason for the type length of 10? If so let's add a small comment as well

Reply:
Yes, code-action won't be readable if there's a lengthy title for it. I'll add a local variable to make the logic clear.